activerecord/test/cases/transaction_callbacks_test.rb

```diff
@@ -290,3 +290,34 @@ def test_after_rollback_called
     assert_equal %w{ after_rollback }, topic.history
   end
 end
+
+class SaveFromAfterCommitBlockTest < ActiveRecord::TestCase
+  self.use_transactional_fixtures = false
+
+  class TopicWithSaveInCallback < ActiveRecord::Base
+    self.table_name = :topics
+    after_commit :cache_topic, :on => :create
+    after_commit :call_update, :on => :update
+    attr_accessor :cached, :record_updated
+
+    def call_update
+      self.record_updated = true
+    end
+
+    def cache_topic
+      unless cached
+        self.cached = true
+        self.save
+      else
+        self.cached = false
+      end
+    end
+  end
```

Comment:
Why does the callback still need a condition for not being called again?
Isn't the purpose of this PR to prevent the callback from being called again after a save?

```
def cache_topic
  self.cached = true
  self.save
end
```
Shouldn't this code work too?